src/App.js

```diff
@@ -1,26 +1,100 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { Component } from 'react';
 import './App.css';
+import contacts from './contacts.json';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends Component { 
+  state = {
+    contactList: contacts.slice(0, 5),
+  };
+
+  addRandom = () => {
+    const copyList = [...this.state.contactList];
+    const random = contacts[Math.floor(Math.random() * contacts.length)];
+    if (!copyList.includes(random)) {
```

Comment:
Please be mindful that this `.includes` works because copyList and random are both derived from contacts, so the objects in these variables have the same object references.

To exemplify that:

1. [{a: 1}].includes({a:1}); // false
2. var arr = [{b:2}]; var arrCopy = arr; arrCopy.includes(arr[0]); // true

https://stackoverflow.com/a/6605700/3563737

Reply:
The logic can be improved. 
If a user clicks the "Add another contact" button, we would like a contact to be added in any case, right?
If a match is found, perhaps another random contact can be retrieved, compared to the copyList and added if there is no match.
This repetitive operation could be stored in a function. For example "getRandomContact".